modules/rds__enum_snapshots/main.py

```diff
@@ -0,0 +1,234 @@
+
+    # For prerequisite modules, try and see if any existing modules return the data that is required for your module before writing that code yourself, that way, session data can stay separated and modular.
+    'prerequisite_modules': [],
+
+    # Module arguments to autocomplete when the user hits tab
+    'arguments_to_autocomplete': ['--regions', '--snapshot-permissions', '--no-include-shared'],
+}
+
+parser = argparse.ArgumentParser(add_help=False, description=module_info['description'])
+
+parser.add_argument(
+    '--regions',
+    required=False,
+    default=None,
+    help='One or more (comma separated) AWS regions in the format "us-east-1". Defaults to all session regions.'
+)
+parser.add_argument(
+    '--snapshot-permissions',
+    required=False,
+    default=False,
+    action='store_true',
+    help='Capture permissions for each found snapshot that belongs to this account. Permissions for snapshots shared to this account are not included. Found permissions will be captured in the database and written to the sessions downloads directory as rds_snapshot_permissions.txt'
+)
+
+parser.add_argument(
+    '--no-include-shared',
+    required=False,
+    default=False,
+    action='store_true',
+    help='Do not include snapshots shared to this account by other accounts.'
+)
+
+
+def main(args, pacu_main):
+    session = pacu_main.get_active_session()
+
+    ###### Don't modify these. They can be removed if you are not using the function.
+    args = parser.parse_args(args)
+    print = pacu_main.print
+    input = pacu_main.input
+    key_info = pacu_main.key_info
+    get_regions = pacu_main.get_regions
+    ######
+
+    rds_data = deepcopy(session.RDS)
+    if 'Snapshots' not in rds_data.keys():
+        rds_data['Snapshots'] = []
+    session.update(pacu_main.database, RDS=rds_data)
+
+    if args.regions is None:
+        regions = get_regions('rds')
+        if regions is None or regions == [] or regions == '' or regions == {}:
+            print('This module is not supported in any regions specified in the current sessions region set. Exiting...')
+            return
+    else:
+        regions = args.regions.split(',')
+
+    client = pacu_main.get_boto3_client('rds', random.choice(regions))
+    now = time.time()
+    all_snaps = []
+    snapshots_csv_data = []
+    shared_by_other_account_snapshots = []
+    snapshot_permissions = {
+        'Public': [],
+        'Shared': {},
+        'Private': [],
+    }
+
+    for region in regions:
+        print('Starting region {} (this may take a while if there are thousands of RDS snapshots)...'.format(region))
+        client = pacu_main.get_boto3_client('rds', region)
+
+        # Start RDS Snapshots in this region
+        count = 0
+        response = None
+        next_token = False
+
+        while (response is None or 'NextToken' in response):
+            if next_token is False:
+                try:
+                    response = client.describe_db_snapshots(
```

Comment:
Calling this method only returns snapshots of DB instances. There also exists snapshot of DB clusters. It would be great if we update this to include DB cluster snapshots as well! Here is a link to the method used to get that data. https://docs.aws.amazon.com/cli/latest/reference/rds/describe-db-cluster-snapshots.html